pkg/configuration/new_asset_store.go

```diff
@@ -4,50 +4,43 @@ import (
 	pb "github.com/buildbarn/bb-remote-asset/pkg/proto/configuration/bb_remote_asset"
 	"github.com/buildbarn/bb-remote-asset/pkg/storage"
 	asset_configuration "github.com/buildbarn/bb-remote-asset/pkg/storage/blobstore"
-	"github.com/buildbarn/bb-storage/pkg/blobstore"
 	blobstore_configuration "github.com/buildbarn/bb-storage/pkg/blobstore/configuration"
 	"github.com/buildbarn/bb-storage/pkg/grpc"
 
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// NewAssetStoreAndCASFromConfiguration creates an Asset Store and
-// BlobAccess for the Content Addressable Storage.
-func NewAssetStoreAndCASFromConfiguration(configuration *pb.AssetCacheConfiguration, grpcClientFactory grpc.ClientFactory, maximumMessageSizeBytes int) (storage.AssetStore, blobstore.BlobAccess, error) {
-	var assetStore storage.AssetStore
-	var contentAddressableStorage blobstore.BlobAccess
+// NewAssetStoreFromConfiguration creates an Asset Store from a
+// configuration and CAS
+func NewAssetStoreFromConfiguration(configuration *pb.AssetCacheConfiguration, contentAddressableStorage blobstore_configuration.BlobAccessInfo, grpcClientFactory grpc.ClientFactory, maximumMessageSizeBytes int) (storage.AssetStore, error) {
+	if configuration == nil {
+		return nil, nil
```

Comment:
Should we ever return nil, nil? Surely if we're not returning an AssetStore then we want an err to say why